test/e2e/util/util.go

```diff
@@ -1855,3 +1855,73 @@ func EnsureSATokenNotMountedUnlessNecessary(t *testing.T, ctx context.Context, c
 		}
 	})
 }
+
```

Comment:
I'd expect this test to be a test on its own pretty similar to RollingUpgradeTest but checking NodePoolUpdatingConfigConditionType instead. See https://github.com/openshift/hypershift/blob/c34a1f6cef0cb41c8a1f83acd4ddf10a4b9e8532/test/e2e/nodepool_rolling_upgrade_test.go